pkg/storage/chunk/cache/redis_client.go

```diff
@@ -84,38 +84,41 @@ func NewRedisClient(cfg *RedisConfig) (*RedisClient, error) {
 }
 
 func deriveEndpoints(endpoint string, lookup func(host string) ([]string, error)) ([]string, error) {
-	if lookup == nil {
-		return nil, fmt.Errorf("lookup function is nil")
-	}
-
-	endpoints := strings.Split(endpoint, ",")
-
-	// no endpoints or multiple endpoints will not need derivation
-	if len(endpoints) != 1 {
-		return endpoints, nil
-	}
-
-	// Handle single configuration endpoint which resolves multiple nodes.
-	host, port, err := net.SplitHostPort(endpoints[0])
-	if err != nil {
-		return nil, fmt.Errorf("splitting host:port failed :%w", err)
-	}
-	addrs, err := lookup(host)
-	if err != nil {
-		return nil, fmt.Errorf("could not lookup host: %w", err)
-	}
-
-	// only use the resolved addresses if they are not all loopback addresses;
-	// multiple addresses invokes cluster mode
-	allLoopback := allAddrsAreLoopback(addrs)
-	if len(addrs) > 1 && !allLoopback {
-		endpoints = nil
-		for _, addr := range addrs {
-			endpoints = append(endpoints, net.JoinHostPort(addr, port))
-		}
-	}
-
-	return endpoints, nil
+    if lookup == nil {
+        return nil, fmt.Errorf("lookup function is nil")
+    }
+
+    endpoints := strings.Split(endpoint, ",")
+
+    // no endpoints or multiple endpoints will not need derivation
+    if len(endpoints) != 1 {
+        return endpoints, nil
+    }
+
+    // Handle single configuration endpoint which resolves multiple nodes.
+    host, port, err := net.SplitHostPort(endpoints[0])
+    if err != nil {
+        // If the port is missing, use the host as is and set a default port, if desired
+        host = endpoints[0]
+        // Example: set a default port, or you can choose to handle this differently
+        port = "6379" // Default Redis port
```

Comment:
this feels like something we should do via a `validateConfig` function before passing the config to the client creation

@dannykopping I think you worked on the caching code the most recently, any thoughts here?

Reply:
Let me know if you would like me to approach this differently. I would be happy to contribute.

Reply:
Following up on this @dannykopping  and @cstyan

Reply:
I would have a new function that's called before `deriveEndpoints` in `NewRedisClient` that logs a warning for every endpoint that doesn't have a port set clarifying that the default port will be used. Alternatively, this new function could be called before `NewRedisClient` in `cache.go` `New`.